File: DemoApp/DemoApp/CustomLayoutNativeAdDetailsViewController.m
//
//  CustomLayoutNativeAdDetailsViewController.m
//  DemoApp
//
//  Created by Glispa GmbH on 20.06.17.
//  Copyright © 2017 Glispa GmbH. All rights reserved.
//

#import "CustomLayoutNativeAdDetailsViewController.h"
#import "NativeBannerView.h"

@interface CustomLayoutNativeAdDetailsViewController ()

@end

@implementation CustomLayoutNativeAdDetailsViewController

- (void)viewDidLoad {
    [super viewDidLoad];
}

- (void)loadAds {
    [super loadAds];
    __weak typeof(self) weakSelf = self;
    [AvocarrotSDK.sharedSDK loadNativeAdWithAdUnitId:self.adUnitId
                                parentViewController:self
                             adViewClassForRendering:[NativeBannerView class]
                                             success:^(UIView * _Nonnull adNativeViewContainer) {
                                                __strong __typeof__(self) sSelf = weakSelf;
                                                 [sSelf displayAdWithAdView:adNativeViewContainer];
                                                 NSLog(@"Nativa ad has loaded!");
                                             } failure:^(AVOError *_Nonnull error) {
                                                 NSLog(@"Native ad loading error: %@", [error localizedDescription]);
                                             }];
}

- (void)prepareHeight {
    self.desiredHeight = [NativeBannerView desiredHeight];
}

@end

File: DemoApp/DemoAppSwift/CustomLayoutNativeAdDetailsViewController.swift
//
//  CustomLayoutNativeAdDetailsViewController.swift
//  DemoApp
//
//  Created by Glispa GmbH on 09.06.17.
//  Copyright © 2017 Glispa GmbH. All rights reserved.
//

import UIKit
import AvocarrotNativeView

@objc(CustomLayoutNativeAdDetailsViewController)
class CustomLayoutNativeAdDetailsViewController: BaseNativeAdDetailsViewController {

    override func loadAds() {
        super.loadAds()

        AvocarrotSDK.shared.loadNativeAd(withAdUnitId: adUnitId,
                                      parentViewController: self,
                                      adViewClassForRendering: NativeBannerView.classForCoder(),
                                      success: { [weak self] (view) in
                                        guard let sSelf = self else {return}
                                        sSelf.displayAd(withAdView: view)
                                        print("Native ad has loaded!")
                                      }, failure: { (error) in
                                        print("Native ad error: \(error.localizedDescription)")
                                      })
    }

    override func prepareHeight() {
        desiredHeight = NativeBannerView.desiredHeight()
    }

}
